tempesta_fw/classifier/frang.c

```diff
@@ -540,18 +583,63 @@ int __fsm_const_state = Frang_Req_0; /* make compiler happy */
 #define __FRANG_FSM_START(st)						\
 switch(st)
 
+#if DEBUG == 3
+
+const char *__state_name_array[] = {
+	"Frang_Req_0",
+
+	"Frang_Req_Hdr_Start",
+	"Frang_Req_Hdr_Method",
+	"Frang_Req_Hdr_UriLen",
+	"Frang_Req_Hdr_FieldDup",
+	"Frang_Req_Hdr_FieldLen",
+	"Frang_Req_Hdr_FieldLenFinal",
+	"Frang_Req_Hdr_Crlf",
+	"Frang_Req_Hdr_Host",
+	"Frang_Req_Hdr_ContentType",
+
+	"Frang_Req_Hdr_NoState",
+
+	"Frang_Req_Body_Start",
+	"Frang_Req_Body_Timeout",
+	"Frang_Req_Body_ChunkCnt",
+	"Frang_Req_Body_Len",
+
+	"Frang_Req_Body_NoState",
+
+	"Frang_Req_Done"
+};
+
+#define __state_name(state) ((state >= 0 && state <= Frang_Req_Done) ?	\
+				__state_name_array[state] :		\
+				"Wrong state")
+
 #define __FRANG_FSM_FINISH()						\
 done:									\
-	TFW_DBG3("Finish FRANG FSM at state %d\n", __fsm_const_state);	\
+	TFW_DBG3("Finish FRANG FSM at state %d = %s\n",			\
+		__fsm_const_state, __state_name(__fsm_const_state));	\
 	TFW_DBG3("Frang return %s\n", r == TFW_PASS ? "PASS" : "BLOCK");\
 	req->frang_st = __fsm_const_state;
 
 #define __FRANG_FSM_STATE(st)						\
 case st:								\
 st: __attribute__((unused))						\
-	TFW_DBG3("enter FRANG FSM at state %d\n", st);			\
+	TFW_DBG3("enter FRANG FSM at state %d = %s\n", st, __state_name(st));\
+	__fsm_const_state = st; /* optimized out to constant */
+
+#else
```

Comment:
You can use a comment to visually distinguish this statement from the rest of the code. Something like that:
```C
#else /* ( DEBUG == 3) */
```